Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs

```diff
@@ -0,0 +1,228 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Runtime.Serialization.Json;
+    using System.Web;
+
+    using Microsoft.ManagementServices.RealTimeDataProcessing.QuickPulseService;
+
+    /// <summary>
+    /// Service client for QPS service.
+    /// </summary>
+    internal sealed class QuickPulseServiceClient : IQuickPulseServiceClient
+    {
+        private const string XMsQpsSubscribedHeaderName = "x-ms-qps-subscribed";
+
+        private const int RetryCount = 3;
+        
+        private readonly string instanceName;
+
+        private readonly string version;
+
+        private readonly TimeSpan timeout = TimeSpan.FromSeconds(3);
+        
+        public QuickPulseServiceClient(Uri serviceUri, string instanceName, string version, TimeSpan? timeout = null)
+        {
+            this.ServiceUri = serviceUri;
+            this.instanceName = instanceName;
+            this.version = version;
+            this.timeout = timeout ?? this.timeout;
+        }
+
+        public Uri ServiceUri { get; }
+
+        public bool? Ping(string instrumentationKey, DateTime timestamp)
+        {
+            MemoryStream bodyStream = this.WritePingData(timestamp);
+
+            var path = string.Format(CultureInfo.InvariantCulture, "ping?ikey={0}", instrumentationKey);
+            HttpWebResponse response = this.SendRequest(WebRequestMethods.Http.Post, path, bodyStream, true);
```

Comment:
I wonder why it is not a `GET` with timestamp in queryString? Why do you need timestamp at all?